### ISSUE TITLE
CTRL-S does not trigger a save in project settings and pipeline settings views.

### ISSUE THREAD

**Issue by fruttasecca** [orchest/orchest#149]
This is inconsistent with the pipeline editor, where CTRL-S will trigger a save.

**Comment by shrikantkarve** [comment]
I am eager to work on this issue. Do let me know how I can get started. @ricklamers @fruttasecca

**Comment by ricklamers** [comment]
@shrikantkarve that's awesome!

I'd start by quickly glancing over how to set up the development environment here: https://orchest.readthedocs.io/en/latest/developer_guide/development_workflow.html

And a good place to start in the code is this line: https://github.com/orchest/orchest/blob/66062df7929756d9089fd71076a773520ce826fe/services/orchest-webserver/client/src/views/SettingsView.jsx#L114

Basically that saveConfig React function needs to be triggered by the shortcut Ctrl + S.

**Comment by ricklamers** [comment]
I would argue Ctrl + S save should work for:

- Environment Build
- Project Settings
- Pipeline Settings
- Global Settings

views. But you can also start with just one of them :)!

**Comment by shrikantkarve** [comment]
Thanks for the info. Let me start understanding the basics.

**Comment by shrikantkarve** [comment]
I did the setup (required a windows update for WSL2). Got the standard orchest installed and working. I am seeing an issue however with "pnpm run dev". Command does not fail but does not complete either.  Any ideas? 

Snippet of "npn run dev" where it stalls below.

lib/javascript/mdc dev: [ts:watch]
lib/javascript/mdc dev: [ts:watch] 6:06:59 PM - Found 0 errors. Watching for file changes.
lib/javascript/mdc dev: [ts:watch] 6:06:59 PM - File change detected. Starting incremental compilation...
lib/javascript/mdc dev: [ts:watch]
lib/javascript/mdc dev: [ts:watch]
lib/javascript/mdc dev: [ts:watch] 6:07:00 PM - Found 0 errors. Watching for file changes.
lib/design-system/package dev: [ts:watch] 6:07:00 PM - Starting compilation in watch mode...
lib/design-system/package dev: [ts:watch]
lib/design-system/package dev: [ts:watch]
lib/design-system/package dev: [ts:watch] 6:07:17 PM - Found 0 errors. Watching for file changes.

**Comment by yannickperrenet** [comment]
> Command does not fail but does not complete either. Any ideas?

Yes. It is not supposed to finish. It starts the vite dev server so that you can issue the `./orchest start --dev` in another terminal and have hot reloading when changing files to ease development (otherwise to see changes you would have to rebuild the Docker container again).

Hope that helps. You have done everything correct so far ;)

**Comment by shrikantkarve** [comment]
Thanks @yannickperrenet. Doc update looks great. I will rebuild the docker container to test the changes, till I get "pnpm run dev" issue resolved.

**Comment by yannickperrenet** [comment]
I am sorry, maybe I was not clear the first time, but the output you are getting when running `pnpm run dev` is the output you are supposed to get (no issue there). Just run `./orchest start --dev` in another terminal and start developing.

The `pnpm run dev` command does not finish and is not supposed to finish.

**Comment by shrikantkarve** [comment]
I tried running "./orchest start --dev" from other terminal but it gives "connection refused" error. Also pnpm command does not show "listening on 3000 port" message, so looks like pnpm is stuck.

**Comment by ricklamers** [comment]
For the record, after a chat with @shrikantkarve on Slack this was resolved as: development environment is not officially support on WSL2.

However, it can be made to run if you run the Docker daemon in WSL2 directly ([Install Docker on Ubuntu](https://docs.docker.com/engine/install/ubuntu/#install-using-the-convenience-script) and `service docker start`).
